snyk/snyk-docker-plugin
# feat: collect node and python app files hierarchy

#### What does this PR do?
https://snyksec.atlassian.net/wiki/x/3oA6o

#### Where should the reviewer start?
Following the logical flow  starting from `lib/analyzer/static-analyzer.ts` will be easier to follow.

#### How should this be manually tested?
Using the Docker registry agent(DRA) (https://github.com/snyk/docker-registry-agent) update the installed Snyk docker plugin version (in the package.json) to be based on the current branch:
`"snyk-docker-plugin": "github:snyk/snyk-docker-plugin#collect_application_files"`.
Run `npm install`.
Run the DRA locally and from a different terminal execute the following command:
```
curl --location 'http://localhost:17500/scan' \
--header 'snyk-source-credentials: <<CREDENTIALS_FOR_CR>>' \
--header 'Content-Type: application/json' \
--data '{
    "target": {
        "name": "<<IMAGE_ON_CR>>"
    },
    "options": {
        "appScan": true,
        "collectApplicationFiles": true
    },
    "callbacks": {
        "resultUrl": "<<RESULT_URL>>",                                   
        "doneUrl": "<<DONE_URL>>",
        "token": "00000000-0000-0000-0000-000000000000"
    }   
}' 
```
The expected output of the given RESULT_URL endpoint should contain the fact type "applicationFiles" as part of the data.
The image should be NodeJS or Python.

#### What are the relevant tickets?
https://snyksec.atlassian.net/browse/RNTM-744
https://snyksec.atlassian.net/browse/RNTM-746
https://snyksec.atlassian.net/browse/RNTM-747
